geopandas/io/arrow.py

```diff
@@ -434,7 +452,20 @@ def _read_parquet(path, columns=None, storage_options=None, **kwargs):
     kwargs["use_pandas_metadata"] = True
     table = parquet.read_table(path, columns=columns, filesystem=filesystem, **kwargs)
 
-    return _arrow_to_geopandas(table)
+    # read metadata separately to get the raw Parquet FileMetaData metadata
+    # (pyarrow doesn't properly exposes those in schema.metadata for files
+    # created by GDAL - https://issues.apache.org/jira/browse/ARROW-16688)
+    try:
+        if filesystem is not None:
+            pa_filesystem = _ensure_arrow_fs(filesystem)
+            with pa_filesystem.open_input_file(path) as source:
+                metadata = parquet.read_metadata(source).metadata
+        else:
+            metadata = parquet.read_metadata(path).metadata
```

Comment:
This is a bit unfortunate, because pyarrow's `parquet.read_metadata` doesn't handle the same path/filesystem combo's as the `read_table` does (as we use it above). I will open an upstream feature request about this, but until then, we will have to manually handle both cases of a filesystem or local file path